server/routes/startOAuthLogin.ts

```diff
@@ -0,0 +1,30 @@
+import passport from 'passport';
+import { Request, Response, NextFunction } from 'express';
+
+import { GITHUB_OAUTH_CALLBACK } from '../config';
+
+const startOAuthLogin = async (models, req: Request, res: Response, next: NextFunction) => {
+  let successRedirect = '/';
+  let failureRedirect = '#!/login';
+  if (req.query.from) {
+    try {
+      const [chain, community] = await Promise.all([
+        models.Chain.findOne({ where: { customDomain: req.query.from } }),
+        models.OffchainCommunity.findOne({ where: { customDomain: req.query.from } }),
+      ]);
+      if (chain || community) {
+        const tokenObj = await models.LoginToken.createForOAuth(req.query.from);
+        successRedirect = `https://${req.query.from}/api/finishOAuthLogin?token=${tokenObj.token}`;
```

Comment:
Should we perform some sort of validation on `from` to avoid phishing etc?

Reply:
We do already by looking up `chain || community`. We could also add a DNS query, or a prompt to the user would make more sense ("you are being logged in on `example.com`"), but if we did the latter then we should also start scoping login tokens to only be valid for a given community.

Reply:
Given that CW is supposed to be a thin interface over Web3, it probably makes more sense for us to make interactions more trustless, for example removing the ability to comment or post unless a valid cryptographic signature is provided.

Reply:
Can you elaborate on what work the more "trustless" mode of interaction would entail?

Reply:
Every interaction on CW requires a signature (other than reactions I suppose). And the signature means that all CW instances can be synced up using Textile/IPFS, Ceramic streams, or some other decentralized backend.

Reply:
I see -- that would be cool at some point, but definitely out of scope.